language/stdlib/modules/validator_set.mvir

```diff
@@ -32,6 +32,42 @@ module ValidatorSet {
         return;
     }
 
+    // Return the size of the current validator set
+    public size(): u64 acquires T {
+        let validator_set_ref: &Self.T;
+
+        validator_set_ref = borrow_global<T>(0x1D8);
+        return Vector.length<Self.ValidatorInfo>(&move(validator_set_ref).validators);
+    }
+
+    // Return true if addr is a current validator
+    public is_validator(addr: address): bool acquires T {
+        let size: u64;
+        let i: u64;
+        let vector_ref: &Vector.T<Self.ValidatorInfo>;
+        let info_ref: &Self.ValidatorInfo;
+
+        vector_ref = &borrow_global<T>(0x1D8).validators;
+        size = Vector.length<Self.ValidatorInfo>(copy(vector_ref));
+        if (copy(size) == 0) {
+            return false;
+        }
+
+        i = 0;
+        info_ref = Vector.borrow<Self.ValidatorInfo>(copy(vector_ref), copy(i));
+        while (copy(i) < copy(size)) {
+            if (*&copy(info_ref).addr == copy(addr)) {
+            // TODO: below doesn't work, looks like a compiler bug
+            //if (*(&Vector.borrow<Self.ValidatorInfo>(copy(vector_ref), copy(i)).addr) == copy(addr)) {
+                return true;
```

Comment:
Shall we comment out the trailing lines as well?

Reply:
Not sure what you mean--we need the `return true`.

Reply:
There seems to be an extra `}` at line 63?

Reply:
This `}` is closing the `{` at line 59.